File: GXpert/GXpert.Web/Modules/ServerTypes/Settings/SmtpColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SmtpRow } from "./SmtpRow";

export interface SmtpColumns {
    Id: Column<SmtpRow>;
    Server: Column<SmtpRow>;
    Port: Column<SmtpRow>;
    Username: Column<SmtpRow>;
    Password: Column<SmtpRow>;
    IsSsl: Column<SmtpRow>;
}

export class SmtpColumns extends ColumnsBase<SmtpRow> {
    static readonly columnsKey = 'Settings.Smtp';
    static readonly Fields = fieldsProxy<SmtpColumns>();
}

File: GXpert/GXpert.Web/Modules/Settings/Smtp/SmtpColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Settings.Columns;

[ColumnsScript("Settings.Smtp")]
[BasedOnRow(typeof(SmtpRow), CheckNames = true)]
public class SmtpColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Server { get; set; }
    public int Port { get; set; }
    public string Username { get; set; }
    public string Password { get; set; }
    public bool IsSsl { get; set; }
}